pkg/storage/bucket/s3/config.go

```diff
@@ -152,6 +153,7 @@ func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.Uint64Var(&cfg.PartSize, prefix+"s3.part-size", 0, "The minimum file size in bytes used for multipart uploads. If 0, the value is optimally computed for each object.")
 	f.BoolVar(&cfg.SendContentMd5, prefix+"s3.send-content-md5", false, "If enabled, a Content-MD5 header is sent with S3 Put Object requests. Consumes more resources to compute the MD5, but may improve compatibility with object storage services that do not support checksums.")
 	f.Var(newBucketLookupTypeValue(s3.AutoLookup, &cfg.BucketLookupType), prefix+"s3.bucket-lookup-type", fmt.Sprintf("Bucket lookup style type, used to access bucket in S3-compatible service. Default is auto. Supported values are: %s.", strings.Join(supportedBucketLookupTypes, ", ")))
+	f.BoolVar(&cfg.DualstackEnabled, prefix+"s3.dualstack-enabled", true, "When enabled, resolves AWS S3 endpoint into the dual IPv4/IPv6 endpoint for the configured region.")
```

Comment:
> [..] into the dual IPv4/IPv6 endpoint for the configured region

I'm not a fan of the wording here, but I took it from [`aws-cli s3` help][1] :)

[1]: https://awscli.amazonaws.com/v2/documentation/api/latest/topic/s3-config.html#use-dualstack-endpoint

Reply:
Great idea to reuse the existing AWS documentation for consistency. I am not sure if their technical writing team review the CLI help but I imagine they do review the documentation pages and it has a similar wording:

https://docs.aws.amazon.com/AmazonS3/latest/userguide/dual-stack-endpoints.html#:~:text=direct%20all%20Amazon%20S3%20requests%20made%20by%20the%20s3%20and%20s3api%20AWS%20CLI%20commands%20to%20the%20dual%2Dstack%20endpoint%20for%20the%20specified%20region

```suggestion
	f.BoolVar(&cfg.DualstackEnabled, prefix+"s3.dualstack-enabled", true, "When enabled, direct all AWS S3 requests to the dual-stack IPv4/IPv6 endpoint for the configured region.")
```

Feel free to use whichever you think is most understandable to a Mimir user.